src/prpy/base/robot.py

```diff
@@ -284,15 +284,33 @@ def ViolatesVelocityLimits(self, traj):
         return False
 
     def _PlanWrapper(self, planning_method, args, kw_args):
-        # Call the planner.
-        traj = planning_method(self, *args, **kw_args)
 
-        # Strip inactive DOFs from the trajectory.
+        # Call the planner.
         config_spec = self.GetActiveConfigurationSpecification()
-        openravepy.planningutils.ConvertTrajectorySpecification(traj, config_spec)
+        result = planning_method(self, *args, **kw_args)
+
+        # Define the post processing steps for the trajectory.
+        def postprocess_trajectory(traj, kw_args):
+
+            # Strip inactive DOFs from the trajectory.
+            openravepy.planningutils.ConvertTrajectorySpecification(
+                traj, config_spec
+            )
+
+            # Optionally execute the trajectory.
+            if 'execute' not in kw_args or kw_args['execute']:
+                return self.ExecuteTrajectory(traj, **kw_args)
+            else:
+                return traj
```

Comment:
We're already "seriously busting the API" if we're making `timeout = 0` immediately throw an exception. Also, there aren't very many cases where we actually use the return value from `ExecuteTrajectory`. The return value primarily exists for debugging purposes (i.e. in case the pre-processing code is broken).

What's the downside to using futures everywhere?